framework/library/data_structures/abstract/linked_list.cpp

```diff
@@ -314,16 +314,89 @@ size_t T3_LinkedListDouble_FindIndexOf(T3_LinkedListDouble *list, T3_NodeDouble
     return -1;
 }
 
-//TODO: Below functions can still optimized!
+//TODO: Check below optimizations
 T3_Node *T3_LinkedList_RemoveAt(T3_LinkedList *list, size_t index) {
-    T3_Node *removed = T3_LinkedList_GetNode(list, index);
-    T3_LinkedList_Remove(list, removed);
+    T3_Assert(list == NULL, "Invalid list")
+    T3_Assert(index >= list->Count, "Invalid index inserted")
+
+    T3_Node *removed = NULL;
+
+    if(index == 0){
+        removed = list->Head;
+        list->Head = removed->Next;
+
+        if(list->Count == 1){
+            list->Tail = NULL;
+        }
+    } else {
```

Comment:
lets say index == 0 and list->Count == 2 . So list->Tail should be null when the head is removed because tail will be head.